pkg/api/pod/util.go

```diff
@@ -1803,3 +1806,19 @@ func workloadRefInUse(podSpec *api.PodSpec) bool {
 
 	return podSpec.WorkloadRef != nil
 }
+
+func containerRestartRulesInUseOnSidecar(oldPodSpec *api.PodSpec) bool {
```

Comment:
Why is this limited to sidecars? If I'm reading correctly, it looks like you allow RestartAllContainers on regular containers too?

Reply:
Oh, maybe this should be checking if any restart rules are set on sidecars, or if the RestartAll rule is used on any container?